### PR TITLE
API to expose gauge metric values on minion

### PR DESCRIPTION
**Pull request by shounakmk219**
# Description

Expose an API that provides the gauge metric value on the minion based on the provided gauge name

Path : `/metrics/gauge/<gaugeName>`
Method : `GET`
Sample response:
```
{
  "gaugeValue": 10,
  "metricExists": true
}
```

### CODE REVIEW

**Review by Jackie-Jiang — COMMENTED** [review]
My gut feel is that we should not use metric value as source of truth..

How about we add an API specific for reading metric values given the metric type and enum value?

**Comment by shounakmk219** [comment]
That makes sense. Will make the changes accordingly.

**Comment by shounakmk219** [comment]
@Jackie-Jiang I explored the option of having an endpoint to get metric values. There were lot of other complications around it hence decided to stick to the existing API just instead of relying on the gauge metric it fetches the active tasks count using `MinionEventObservers`

**Comment by shounakmk219** [comment]
@Jackie-Jiang based on your inputs updated the code to expose just the gauge values for now.

File: pinot-common/src/main/java/org/apache/pinot/common/metrics/AbstractMetrics.java [hunk 1]
```diff
@@ -494,6 +494,16 @@ public void addValueToGlobalGauge(final G gauge, final long unitCount) {
     }
   }
 
+  /**
+   * Get the gauge metric value for the provided gauge name
+   * @param gaugeName gauge name
+   * @return gauge value. If gauge is not present return null.
+   */
+  public Long getGaugeValue(final String gaugeName) {
```
Comment (Jackie-Jiang) [inline pinot-common/src/main/java/org/apache/pinot/common/metrics/AbstractMetrics.java:502]: Annotate it with `@Nullable`

**Review by Jackie-Jiang — APPROVED** [review]

File: pinot-minion/src/main/java/org/apache/pinot/minion/api/resources/PinotMinionMetricsResource.java [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+public class PinotMinionMetricsResource {
+  public static final String METRIC_EXISTS = "metricExists";
+  public static final String GAUGE_VALUE = "gaugeValue";
+  private final MinionMetrics _minionMetrics = MinionMetrics.get();
+
+  @GET
+  @Path("/gauge/{gaugeName}")
+  @Produces(MediaType.APPLICATION_JSON)
+  @ApiOperation("Get gauge value for the provided minion gauge name")
+  public String getMinionGaugeValue(@ApiParam(value = "Gauge name") @PathParam("gaugeName") String gaugeName)
```
Comment (Jackie-Jiang) [inline pinot-minion/src/main/java/org/apache/pinot/minion/api/resources/PinotMinionMetricsResource.java:56]: I guess we should also add access control. cc @soumitra-st

Comment (shounakmk219) [reply pinot-minion/src/main/java/org/apache/pinot/minion/api/resources/PinotMinionMetricsResource.java:56]: I skipped it as the task progress APIs on minion doesn’t have it either. Will check with Soumitra.

Comment (soumitra-st) [reply pinot-minion/src/main/java/org/apache/pinot/minion/api/resources/PinotMinionMetricsResource.java:56]: @shounakmk219 , is it feasible to make /gauge API work for all possible gauges in the system, not just minion? Let's add AUTHZ for this as Jackie suggested.

**Pull request merged**